lakefile.lean

```diff
@@ -171,32 +188,45 @@ module_facet docs (mod) : FilePath := do
               args := #["single", "--build", buildDir.toString, mod.name.toString, srcUri]
               env := ← getAugmentedEnv
             }
-          return docFile
+          return DepSet.mk #[docFile] docDeps
 
--- TODO: technically speaking this target does not show all file dependencies
-def coreTarget (component : Lean.Name) : FetchM (Job FilePath) := do
+def coreTarget (component : Lean.Name) : FetchM (Job <| DepSet FilePath) := do
   let exeJob ← «doc-gen4».fetch
   let bibPrepassJob ← bibPrepass.fetch
   let dataPath := (← getRootPackage).buildDir / "doc-data"
+  let manifestFile := (← getRootPackage).buildDir / s!"{component}-manifest.json"
   let dataFile := dataPath / s!"declaration-data-{component}.bmp"
   let buildDir := (← getRootPackage).buildDir
   bibPrepassJob.bindM fun _ => do
-    exeJob.mapM fun exeFile  => do
-      buildFileUnlessUpToDate' dataFile do
+    exeJob.mapM fun exeFile => do
+      -- TODO: should fold in the dataFile into the trace too? This writes out both.
+      buildFileUnlessUpToDate' manifestFile do
         proc {
           cmd := exeFile.toString
-          args := #["genCore", component.toString, "--build", buildDir.toString]
+          args := #["genCore", component.toString,
+            "--build", buildDir.toString,
+            "--manifest", manifestFile.toString]
           env := ← getAugmentedEnv
         }
-      return dataFile
+      addTrace (← computeTrace dataFile)
+      match Lean.Json.parse <| ← IO.FS.readFile manifestFile with
```

Comment:
Strictly speaking I think I need to write relative paths to the manifest, so that the build directory can be relocated without invalidating the outputs.